kedro_azureml/datasets/folder_dataset.py

```diff
@@ -43,3 +59,87 @@ def __init__(
                 f"underlying dataset. Please remove '{VERSIONED_FLAG_KEY}' flag from "
                 f"the dataset definition."
             )
+
+    @property
+    def path(self) -> str:
+        # For local runs we want to replicate the folder structure of the remote dataset.
+        # Otherwise kedros versioning would version at the file/folder level and not the
+        # AzureML dataset level
+        if self._local_run:
+            return (
+                Path(self.folder)
+                / self._azureml_dataset
+                / self.resolve_load_version()
+                / Path(self._dataset_config[self._filepath_arg])
+            )
+        else:
+            return Path(self.folder) / Path(self._dataset_config[self._filepath_arg])
+
+    @property
+    def download_path(self) -> str:
+        # Because `is_dir` and `is_file` don't work if the path does not
+        # exist, we use this heuristic to identify paths vs folders.
+        if self.path.suffix != "":
+            return str(self.path.parent)
```

Comment:
Shouldn't this be
```suggestion
            return str(
                Path(self.folder)
                / self._azureml_dataset
                / self.resolve_load_version()
            )
```
to support `filepath` entries with multiple levels?

Reply:
in the cases where we need the download path (`self._local_run == True`) `self.path` already includes the dataset and version. So only the parent (version for file/folder at root and parent folder for subfolder) is needed.

Reply:
I think I covered these in the test cases as well

Reply:
Ah, maybe I misunderstood then how the downloading works. I guess it then only downloads the file directly into the folder you specify, without creating the folder structure as in the dataset?

Reply:
correct, that confused me at the beginning as well!